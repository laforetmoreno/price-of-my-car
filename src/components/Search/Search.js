import React from 'react';
import { arrayOf, func, string, objectOf, bool } from 'prop-types';

import Select from '../Select';

import style from './Search.module.scss';

const Search = ({
  brandsOptions = [],
  modelsOptions = [],
  yearsOptions = [],
  versionOptions = [],
  onBrandChange,
  onModelChange,
  onYearChange,
  onVersionsChange,
  disableSelect,
}) => {
  const formattedVersionOptions = () =>
    versionOptions.map(option => ({
      value: option.versionId,
      label: option.version,
    }));

  return (
    <div className={style.wrapper}>
      <h4>MARCA E MODELO DO CARRO</h4>
      <Select
        className={style.select}
        placeholder="Selecione uma marca"
        options={brandsOptions}
        onChange={onBrandChange}
      />
      <Select
        className={style.select}
        disabled={disableSelect.models}
        onChange={onModelChange}
        placeholder="Selecione um modelo"
        options={modelsOptions}
      />
      <h4>ANO E VERSÃO</h4>
      <Select
        disabled={disableSelect.year}
        className={style.select}
        placeholder="Selecione um ano"
        onChange={onYearChange}
        options={yearsOptions}
      />
      <Select
        className={style.select}
        disabled={disableSelect.versions}
        placeholder="Selecione a versão do modelo"
        onChange={onVersionsChange}
        options={formattedVersionOptions()}
      />
    </div>
  );
};

Search.propTypes = {
  brandsOptions: arrayOf(string),
  modelsOptions: arrayOf(string),
  yearsOptions: arrayOf(string),
  versionOptions: arrayOf(string),
  disableSelect: objectOf(bool),
  onBrandChange: func,
  onModelChange: func,
  onYearChange: func,
  onVersionsChange: func,
};

export default Search;
