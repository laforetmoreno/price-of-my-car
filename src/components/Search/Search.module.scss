.wrapper {
  display: flex;
  flex-direction: column;
}

.select {
  margin: 5px 0;
}
