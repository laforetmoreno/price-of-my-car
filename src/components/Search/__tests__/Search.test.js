import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../Search';

describe('<Search />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Search
          brandsOptions={[]}
          modelsOptions={[]}
          yearsOptions={[]}
          versionOptions={[]}
          disableSelect={{}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
