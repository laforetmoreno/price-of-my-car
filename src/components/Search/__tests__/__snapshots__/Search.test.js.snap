// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
Array [
  <div
    className="wrapper collapse"
  >
    <div
      className="wrapper"
    >
      <h4
        className="blockTitle"
      >
        Marca e modelo do carro
      </h4>
      <select
        className="wrapper select undefined"
        data-testid="select"
        defaultValue="default"
        name="brand"
        onChange={[Function]}
      >
        <option
          value="default"
        >
          Selecione uma marca
        </option>
      </select>
      <select
        className="wrapper select undefined"
        data-testid="select"
        defaultValue="default"
        name="model"
        onChange={[Function]}
      >
        <option
          value="default"
        >
          Selecione um modelo
        </option>
      </select>
      <h4
        className="blockTitle"
      >
        Ano e versão
      </h4>
      <select
        className="wrapper select undefined"
        data-testid="select"
        defaultValue="default"
        name="year"
        onChange={[Function]}
      >
        <option
          value="default"
        >
          Selecione o ano
        </option>
      </select>
      <select
        className="wrapper select undefined"
        data-testid="select"
        defaultValue="default"
        name="versionId"
        onChange={[Function]}
      >
        <option
          value="default"
        >
          Selecione a versão do modelo
        </option>
      </select>
    </div>
  </div>,
  <button
    className="btn"
    type="button"
  >
    <img
      alt="Arrow"
      src="arrow.svg"
    />
  </button>,
]
`;
